// @typed/logger/esm/createConsoleLogger.js
import { Pure } from '@typed/env';
/**
 * Create a logger that wraps the console.
 */
export function createConsoleLogger({ logLevel, clock }) {
    const timers = {};
    const logger = {
        log: (msg) => Pure.fromIO(() => {
            if (logLevel > 0 /* OFF */) {
                console.log(msg);
            }
        }),
        error: (msg) => Pure.fromIO(() => {
            if (logLevel > 0 /* OFF */) {
                console.error(msg);
            }
        }),
        clear: () => Pure.fromIO(() => {
            if (logLevel > 0 /* OFF */ && logLevel < 3 /* DEBUG */) {
                console.clear();
            }
        }),
        info: (msg) => Pure.fromIO(() => {
            if (logLevel >= 2 /* INFO */) {
                console.info(msg);
            }
        }),
        debug: (msg) => Pure.fromIO(() => {
            if (logLevel >= 3 /* DEBUG */) {
                console.debug(msg);
            }
        }),
        timeStart: (label) => Pure.fromIO(() => {
            if (logLevel < 3 /* DEBUG */) {
                return;
            }
            timers[label] = clock.currentTime();
        }),
        timeEnd: (label) => Pure.fromIO(() => {
            if (logLevel < 3 /* DEBUG */ || !timers[label]) {
                return -1;
            }
            const startTime = timers[label];
            const elapsed = clock.currentTime() - startTime;
            delete timers[label];
            console.debug(`${label}: ${elapsed}ms`);
            return elapsed;
        }),
    };
    return logger;
}
