// @typed/node/esm/figures/index.js
export const tick = '✔';
export const cross = '✖';
export const star = '★';
export const square = '▇';
export const squareSmall = '◻';
export const squareSmallFilled = '◼';
export const play = '▶';
export const circle = '◯';
export const circleFilled = '◉';
export const circleDotted = '◌';
export const circleDouble = '◎';
export const circleCircle = 'ⓞ';
export const circleCross = 'ⓧ';
export const circlePipe = 'Ⓘ';
export const circleQuestionMark = '?⃝';
export const bullet = '●';
export const dot = '․';
export const line = '─';
export const ellipsis = '…';
export const pointer = '❯';
export const pointerSmall = '›';
export const info = 'ℹ';
export const warning = '⚠';
export const hamburger = '☰';
export const smiley = '㋡';
export const mustache = '෴';
export const heart = '♥';
export const arrowUp = '↑';
export const arrowDown = '↓';
export const arrowLeft = '←';
export const arrowRight = '→';
export const radioOn = '◉';
export const radioOff = '◯';
export const checkboxOn = '☒';
export const checkboxOff = '☐';
export const checkboxCircleOn = 'ⓧ';
export const checkboxCircleOff = 'Ⓘ';
export const questionMarkPrefix = '?⃝';
export const oneHalf = '½';
export const oneThird = '⅓';
export const oneQuarter = '¼';
export const oneFifth = '⅕';
export const oneSixth = '⅙';
export const oneSeventh = '⅐';
export const oneEighth = '⅛';
export const oneNinth = '⅑';
export const oneTenth = '⅒';
export const twoThirds = '⅔';
export const twoFifths = '⅖';
export const threeQuarters = '¾';
export const threeFifths = '⅗';
export const threeEighths = '⅜';
export const fourFifths = '⅘';
export const fiveSixths = '⅚';
export const fiveEighths = '⅝';
export const sevenEighths = '⅞';
