// @typed/logger/cjs/createTestLogger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTestLogger = void 0;
const env_1 = require("@typed/env");
const objects_1 = require("@typed/objects");
/**
 * Create a logger suitable for testing environments.
 */
function createTestLogger({ logLevel, clock }) {
    const timers = {};
    const logs = [];
    const logger = {
        log: (msg) => env_1.Pure.fromIO(() => {
            if (logLevel > 0 /* OFF */) {
                logs.push({ type: 'log', message: msg });
            }
        }),
        error: (msg) => env_1.Pure.fromIO(() => {
            if (logLevel > 0 /* OFF */) {
                logs.push({ type: 'error', message: msg });
            }
        }),
        clear: () => env_1.Pure.fromIO(() => {
            if (logLevel > 0 /* OFF */ && logLevel < 3 /* DEBUG */) {
                logs.push({ type: 'clear' });
            }
        }),
        info: (msg) => env_1.Pure.fromIO(() => {
            if (logLevel >= 2 /* INFO */) {
                logs.push({ type: 'info', message: msg });
            }
        }),
        debug: (msg) => env_1.Pure.fromIO(() => {
            if (logLevel >= 3 /* DEBUG */) {
                logs.push({ type: 'debug', message: msg });
            }
        }),
        timeStart: (label) => env_1.Pure.fromIO(() => {
            if (logLevel < 3 /* DEBUG */) {
                return;
            }
            const time = clock.currentTime();
            timers[label] = time;
            logs.push({ type: 'timeStart', label, time });
        }),
        timeEnd: (label) => env_1.Pure.fromIO(() => {
            if (logLevel < 3 /* DEBUG */) {
                return -1;
            }
            const time = clock.currentTime();
            const startTime = timers[label];
            const elapsed = startTime - time;
            delete timers[label];
            logs.push({ type: 'timeEnd', label, time });
            return elapsed;
        }),
    };
    return {
        logger,
        getLogs: () => objects_1.clone(logs),
    };
}
exports.createTestLogger = createTestLogger;
