// @typed/node/cjs/figures/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sevenEighths = exports.fiveEighths = exports.fiveSixths = exports.fourFifths = exports.threeEighths = exports.threeFifths = exports.threeQuarters = exports.twoFifths = exports.twoThirds = exports.oneTenth = exports.oneNinth = exports.oneEighth = exports.oneSeventh = exports.oneSixth = exports.oneFifth = exports.oneQuarter = exports.oneThird = exports.oneHalf = exports.questionMarkPrefix = exports.checkboxCircleOff = exports.checkboxCircleOn = exports.checkboxOff = exports.checkboxOn = exports.radioOff = exports.radioOn = exports.arrowRight = exports.arrowLeft = exports.arrowDown = exports.arrowUp = exports.heart = exports.mustache = exports.smiley = exports.hamburger = exports.warning = exports.info = exports.pointerSmall = exports.pointer = exports.ellipsis = exports.line = exports.dot = exports.bullet = exports.circleQuestionMark = exports.circlePipe = exports.circleCross = exports.circleCircle = exports.circleDouble = exports.circleDotted = exports.circleFilled = exports.circle = exports.play = exports.squareSmallFilled = exports.squareSmall = exports.square = exports.star = exports.cross = exports.tick = void 0;
exports.tick = '✔';
exports.cross = '✖';
exports.star = '★';
exports.square = '▇';
exports.squareSmall = '◻';
exports.squareSmallFilled = '◼';
exports.play = '▶';
exports.circle = '◯';
exports.circleFilled = '◉';
exports.circleDotted = '◌';
exports.circleDouble = '◎';
exports.circleCircle = 'ⓞ';
exports.circleCross = 'ⓧ';
exports.circlePipe = 'Ⓘ';
exports.circleQuestionMark = '?⃝';
exports.bullet = '●';
exports.dot = '․';
exports.line = '─';
exports.ellipsis = '…';
exports.pointer = '❯';
exports.pointerSmall = '›';
exports.info = 'ℹ';
exports.warning = '⚠';
exports.hamburger = '☰';
exports.smiley = '㋡';
exports.mustache = '෴';
exports.heart = '♥';
exports.arrowUp = '↑';
exports.arrowDown = '↓';
exports.arrowLeft = '←';
exports.arrowRight = '→';
exports.radioOn = '◉';
exports.radioOff = '◯';
exports.checkboxOn = '☒';
exports.checkboxOff = '☐';
exports.checkboxCircleOn = 'ⓧ';
exports.checkboxCircleOff = 'Ⓘ';
exports.questionMarkPrefix = '?⃝';
exports.oneHalf = '½';
exports.oneThird = '⅓';
exports.oneQuarter = '¼';
exports.oneFifth = '⅕';
exports.oneSixth = '⅙';
exports.oneSeventh = '⅐';
exports.oneEighth = '⅛';
exports.oneNinth = '⅑';
exports.oneTenth = '⅒';
exports.twoThirds = '⅔';
exports.twoFifths = '⅖';
exports.threeQuarters = '¾';
exports.threeFifths = '⅗';
exports.threeEighths = '⅜';
exports.fourFifths = '⅘';
exports.fiveSixths = '⅚';
exports.fiveEighths = '⅝';
exports.sevenEighths = '⅞';
